prov/gni/src/gnix_fabric.c

```diff
@@ -59,6 +59,12 @@
 const char gnix_fab_name[] = "gni";
 const char gnix_dom_name[] = "/sys/class/gni/kgni0";
 
+/* TODO: this will need to be adjustable - probably set in GNI_INI*/
+uint32_t gnix_def_max_nics_per_ptag = 4;
+uint32_t gnix_nics_per_ptag[GNI_PTAG_USER_END] = {0};
```

Comment:
Seems like this could be an array of atomics, getting rid of the corresponding lock.  Is there another anticipated use for this field that requires a `uint32_t` and/or a lock?